streaming/src/test/scala/org/apache/spark/streaming/ReceivedBlockHandlerSuite.scala

```diff
@@ -204,26 +222,26 @@ class ReceivedBlockHandlerSuite
     sparkConf.set("spark.storage.unrollFraction", "0.4")
     // Block Manager with 12000 * 0.4 = 4800 bytes of free space for unroll
     blockManager = createBlockManager(12000, sparkConf)
+    // This block is way too large to possibly be cached in memory:
+    def hugeBlock: IteratorBlock = IteratorBlock(List.fill(100)(new Array[Byte](1000)).iterator)
 
     // there is not enough space to store this block in MEMORY,
     // But BlockManager will be able to serialize this block to WAL
     // and hence count returns correct value.
```

Comment:
@dibbhatt, I'm confused because it seems like your comment says that we should fail a job if blocks cannot be persisted because without that persistence the job will not work correctly even if the WAL is enabled. However, that claim seems to be contradicted by the comment describing this test case, which seems to suggest that this job should succeed despite the block being far too large to be successfully stored. In the old test case, however, the block appeared to be too small and actually _was_ being stored in memory, meaning that this comment wasn't describing the actual behavior of the test.